Spanner/src/Transaction.php

```diff
@@ -615,6 +648,19 @@ public function commit(array $options = [])
             throw new \BadMethodCallException('The transaction cannot be committed because it is not active');
         }
 
+        if (is_array($this->transactionId) and isset($this->transactionId['begin'])) {
```

Comment:
We need to enforce typing so that we know `$transactionId` is an array (or a string). It should not be able to be both.

Reply:
I think we should go for a new attribute then.

Reply:
agreed